tests/jsonutils/test_flattener.py

```diff
@@ -417,9 +417,8 @@ def test_flattener_full_example():
 @pytest.mark.parametrize("test_schema", CIRCULAR_SCHEMAS)
 def test_circular_reference(test_schema):
     flattener = JsonSchemaFlattener(test_schema)
-    with pytest.raises(CircularRefError) as excinfo:
-        flattener.flatten_schema()
-    assert "#/properties/a" in str(excinfo.value)
+    flattened_schema = flattener.flatten_schema()
+    assert flattened_schema
```

Comment:
can we add a more thorough unit test here? As in, let's actually check that the circular ref is calculated properly